packages/grafana-ui/src/components/Table/BarGaugeCell.tsx

```diff
@@ -51,6 +63,7 @@ export const BarGaugeCell: FC<TableCellProps> = (props) => {
         itemSpacing={1}
         lcdCellWidth={8}
         displayMode={barGaugeMode}
+        cellLinkFn={getCellLinkFn}
```

Comment:
Instead of extending the `BarGauge` API, I would suggest wrapping it in a link. Take a look at `/app/plugins/panel/bargauge/BarGaugePanel.tsx` - the BarGauge component is there wrapped with a context menu. In the case of a table cell, I think we just need to wrap it BarGauge in a link. The entire cell will be clickable then which in my opinion is fine.

Moreover, the `cellLinkFn` property name is syntactically binding the BarGauge with a table cell, while BarGauge can be used in any other context (i.e. standalone panel).

Reply:
Thank you for the suggestion Dominik! I was also thinking that the new property on `BarGauge` wasn't the best approach but didn't think of your solution. It's much better. Will work on it right away!